php/libraries/Utility.class.inc

```diff
@@ -214,12 +214,37 @@ class Utility
         $factory = \NDB_Factory::singleton();
         $DB      = $factory->database();
         // get the list of projects
-        $projects = $DB->pselect("SELECT * FROM Project", array());
-        $project  = array();
-        foreach ($projects as $row) {
-            $project[$row['ProjectID']] = $row['Name'];
-        }
-        return $project;
+        $projects = $DB->pselectColWithIndexKey(
+            "SELECT ProjectID, Name FROM Project",
+            array(),
+            'ProjectID'
+        ) ?? array();
+        return $projects;
+    }
+
+    /**
+     * Returns a list of study Projects associated with a subproject
+     *
+     * @param int $subprojectID ID of the subproject for which you want to get the
+     *                          associated projects
+     *
+     * @return array an associative array("projectID" => "project title")
+     *
+     * @access public
+     */
+    static function getProjectListForSubproject(int $subprojectID): array
```

Comment:
This is not used.
And it should probably be in the `Subproject` class methods. (I know there is no such thing)

Reply:
removed